spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala

```diff
@@ -101,6 +106,26 @@ private[sharing] abstract class RemoteDeltaFileIndexBase(
       partitionFilters)
     new Column(rewrittenFilters.reduceLeftOption(And).getOrElse(Literal(true)))
   }
+
+  // Converts the specified SQL expressions to a json predicate.
+  //
+  // If the conversion fails, returns a None, which will imply that we will
+  // not perform json predicate based filtering.
+  protected def convertToJsonPredicate(partitionFilters: Seq[Expression]) : Option[String] = {
+    try {
+      val op = OpConverter.convert(partitionFilters)
```

Comment:
question: do you need validate() here?

Reply:
We will validate on server side so that clients don't need to make these assumptions.
The server may enforce other limits such as message size and tree depth.

Reply:
then validate() is mainly used for testing purpose?

Reply:
The OSS delta sharing server will also use it.